src/storage/v2/delta.hpp

```diff
@@ -154,9 +154,14 @@ struct Delta {
   struct RemoveInEdgeTag {};
   struct RemoveOutEdgeTag {};
 
-  Delta(DeleteDeserializedObjectTag /*tag*/, std::atomic<uint64_t> *timestamp,
-        const std::optional<std::string> &old_disk_key)
-      : action(Action::DELETE_DESERIALIZED_OBJECT), timestamp(timestamp), command_id(0), old_disk_key(old_disk_key) {}
+  // DELETE_DESERIALIZED_OBJECT is used to load data from disk committed by past txs.
+  // Because of this object was created in past txs, we create timestamp by ourselves inside instead of having it from
+  // current tx. This timestamp we got from RocksDB timestamp stored in key.
+  Delta(DeleteDeserializedObjectTag /*tag*/, uint64_t ts, const std::optional<std::string> &old_disk_key)
```

Comment:
Is `DELETE_DESERIALIZED_OBJECT` only used by on disk? Because if not this will break in memory.

Reply:
DELETE_DESERIALIZED_OBJECT is used only for disk storage. We create it when object is read from disk.

Reply:
Also this change is already committed in my [PR](https://github.com/memgraph/memgraph/pull/1179)